riscv_isa.ac

```diff
@@ -223,5 +225,33 @@ ISA_CTOR(riscv){
 
   REMU.set_asm("REMU %reg %reg %reg", rd, rs1, rs2);
   REMU.set_decoder(funct7 = 0x01, funct3 = 0x7, op = 0x33);
+
+  //RV32A
+  AMOSWAP_W.set_asm("AMOSWAP.W %reg, %reg, %reg", rd, rs1, rs2);
+  AMOSWAP_W.set_decoder(funct7 = 0x04, funct3 = 0x2, op = 0x2f);  
```

Comment:
Is there a reference document where we can find those values (encoding)? Put its name and/or link as a comment at the top of this file.